dashboard/app/models/contact_rollups_pardot_memory.rb

```diff
@@ -263,18 +263,27 @@ def self.query_updated_contacts
 
   # Deletes prospects from Pardot API that have been marked for deletion
   # by the account deletion process.
-  def self.delete_pardot_prospects
+  def self.delete_pardot_prospects(is_dry_run: false)
     emails = ContactRollupsPardotMemory.
                where.not(marked_for_deletion_at: nil).
                pluck(:email)
 
     deleted_emails = []
-    emails.each do |email|
-      deleted_emails << email if PardotV2.delete_prospects_by_email(email)
+    unless is_dry_run
+      emails.each do |email|
+        deleted_emails << email if PardotV2.delete_prospects_by_email(email)
+      end
     end
 
     # clean-up step to delete rows once they've been deleted from Pardot
-    ContactRollupsPardotMemory.where(email: deleted_emails).delete_all if deleted_emails.present?
+    ContactRollupsPardotMemory.where(email: deleted_emails).delete_all if deleted_emails.present? && !is_dry_run
+
+    CDO.log.info "[Dry run] #{emails.length} total prospects to be deleted from Pardot." if is_dry_run
```

Comment:
nit: these statements could be re-arranged to make them easier to read and less error-prone
```ruby
if is_dry_run
  CDO.log.info "...."
else
   emails.each do ... end
   ContactRollupsPardotMemory.where(...).delete_all
end
```